services/graph/pkg/service/v0/rolemanagement.go

```diff
@@ -0,0 +1,68 @@
+package svc
+
+import (
+	"fmt"
+	"net/http"
+	"net/url"
+
+	"github.com/cs3org/reva/v2/pkg/conversions"
+	"github.com/go-chi/chi/v5"
+	"github.com/go-chi/render"
+	libregraph "github.com/owncloud/libre-graph-api-go"
+	"github.com/owncloud/ocis/v2/services/graph/pkg/service/v0/errorcode"
+)
+
+// GetRoleDefinitions a list of permission roles than can be used when sharing with users or groups
+func (g Graph) GetRoleDefinitions(w http.ResponseWriter, r *http.Request) {
+	logger := g.logger.SubloggerWithRequestID(r.Context())
+	roleList, err := getRoleDefinitionList(g.config.FilesSharing.EnableResharing)
+	if err != nil {
+		logger.Debug().Err(err).Msg("could not get RoleDefinitions")
+		errorcode.GeneralException.Render(w, r, http.StatusInternalServerError, "could not get RoleDefinitions")
+		return
+	}
+	render.Status(r, http.StatusOK)
+	render.JSON(w, r, roleList)
+}
+
+// GetRoleDefinition a permission role than can be used when sharing with users or groups
+func (g Graph) GetRoleDefinition(w http.ResponseWriter, r *http.Request) {
+	logger := g.logger.SubloggerWithRequestID(r.Context())
+	roleID, err := url.PathUnescape(chi.URLParam(r, "roleID"))
+	if err != nil {
+		logger.Debug().Err(err).Str("roleID", chi.URLParam(r, "roleID")).Msg("could not get roleID: unescaping is failed")
+		errorcode.InvalidRequest.Render(w, r, http.StatusBadRequest, "unescaping role id failed")
+		return
+	}
+	role, err := getRoleDefinition(roleID, g.config.FilesSharing.EnableResharing)
+	if err != nil {
+		logger.Debug().Str("roleID", roleID).Msg("could not get role: not found")
+		errorcode.ItemNotFound.Render(w, r, http.StatusNotFound, err.Error())
+		return
+	}
+	render.Status(r, http.StatusOK)
+	render.JSON(w, r, role)
+}
+
+func getRoleDefinitionList(resharing bool) ([]*libregraph.UnifiedRoleDefinition, error) {
```

Comment:
Whats the point in having this return an `error`? Afais it can't error....

Reply:
removed